### DISCUSSION TITLE
Stereo Processing Width: Auto

### DISCUSSION THREAD

**Discussion by disk02** [nagadomi/nunif#193] [Ideas]
The Stereo Processing Width option has really improved the quality of stereo generation. However sometimes an image set can have varying width and is a pain to process manually. I'd love to have an Auto option which checks the resolution of each image before processing to set the correct width. This may not make sense for video but in that case you would just manually set the width.

**Comment by nagadomi** [comment]
if 1920 or 1280 is specified and the input image is smaller than that, the width of the input image is automatically used.
Not sure what the correct width means, but if it is the width of the input image, it is already supported.


---
Here are the specifications for `Stereo Processing Width`.

In my intention, `Stereo Processing Width` is a feature used to **downscale** the depth image when the width of the depth image is greater than 1920 for **yml import**.
But you can also use it as an upscaler if you consider it useful for something.

- When `Default` is specified, the depth images will not be resized (do nothing).
- If any numerical value is specified, the depth images will be resized to the specified size before being input into `row_flow_v3` model. If the depth image is larger than the specified size, it will be downscaled; if it is smaller, it will be upscaled.
- `Stereo Processing Width` does not upscale depth to a size larger than the input image size.
-  It is a wrong choice to specify larger than 1920 for `Stereo Processing Width`. Because `row_flow_v3` does not support width > 1920.

**Reply by disk02** [reply]
I appreciate the clarity. I misunderstood the feature. My guess was that it was related to the trying to prevent distortions sometimes found in landscape or wide images because of how DepthAnything resizes the shorter side of an image down to the depth resolution which can pinch an image.

**Reply by nagadomi** [reply]
Actually, this affects the results of the stereo generation.
In the case of resize, the depth image is resized to the correct aspect ratio before being input into row_flow_v3. In the non-resize case, the result of row_flow_v3 (transformation matrix for image warp) is resized afterwards.
I am not sure if that will be better or worse in the upscale case. However, processing speed will be slow for sure.

**Comment by diverswan9** [comment]
What is the 'default' value for stereo width?

 (what value would be entered for cli?)

**Reply by nagadomi** [reply]
>When Default is specified, the depth images will not be resized (do nothing).

In CLI, if `--stereo-width` option not specified, it will be None (=Default in GUI).
(What you want is not to use `--stereo-width` )

>--stereo-width STEREO_WIDTH
>                        input width for row_flow_v3/row_flow_v2 model (default: None)

**Reply by diverswan9** [reply]
ok... so don't put anything in for a value (leave it blank?) I tried putting in 'none' and it threw an error. I also tried 'Default'. It didn't like that either.

**Reply by nagadomi** [reply]
Do not include `--stereo-width` option in the command line.

**Reply by diverswan9** [reply]
ok
